### PR TITLE
Change logging so that the messages appear once on thread status change

### LINKED ISSUE #21: Stop reccurring messages for hanging threads
If a thread hangs, this incedente should be reported once only. If the hanging thread awakes, then this event should be logged too. No repeating messages should be logged in the interval between these two events.
This behaviour is proposed to solve the problem of swamping messages discussed in #15.

### PR DESCRIPTION
**Pull request by artemrizhov**
Closes #21.

### CODE REVIEW

**Comment by niccokunzmann** [comment]
You reordered the functions so I can not really see what changed. Would you like to merge it like this or create two commits?

File: hanging_threads.py [hunk 1]
```diff
@@ -75,62 +147,28 @@ def frame2string(frame):
...
 
-    seconds_frozen - How much time should thread hang to activate printing stack trace - default(10)
-    tests_per_second - How much tests per second should be done for hanging threads - default(10)
-    """
-    thread = StoppableThread(target=monitor, args=(seconds_frozen,
-                                                   tests_per_second))
-    thread.daemon = True
-    thread.start()
-    return thread
+def write_log(title, message=''):
```
Comment (niccokunzmann) [inline hanging_threads.py:157]: I like this: It is a basis of setting a log

**Review by niccokunzmann — APPROVED** [review]

File: hanging_threads.py (hunk 1 above)
Comment (artemrizhov) [reply hanging_threads.py:157]: I thought somebody may want to replace this function with own implementation to write messages to another destination in another format.

**Comment by artemrizhov** [comment]
Ok, no problem. I just found that we lost structure.

**Comment by artemrizhov** [comment]
I've changed the order back so that you can review the changes. I'll create another PR for with reordered functions then.

File: hanging_threads.py [hunk 2]
```diff
@@ -91,35 +99,65 @@ def monitor(seconds_frozen, tests_per_second):
...
+                last_change_time = old_threads[thread_id]['time']
+                thread_data['time'] = last_change_time
+                # Check if this is a new hanging thread.
+                if (thread_id not in hanging_threads and
+                        last_change_time < then):
+                    # Gotcha!
+                    hanging_threads.add(thread_id)
+                    # Report the hanged thread.
+                    log_hanged_thread(thread_id, frame)
         old_threads = new_threads
```
Comment (artemrizhov) [inline hanging_threads.py:141]: I have to say that the `monitor()` function becomes too large. However I have not found simple elegant way to divide it into small chunks. I think it should be rewritten in FP or OOP style. But this is probably a job for another task. For now I just want to make it usable for me and others to let us all fix the hanging threads problem in our apps :)

Comment (niccokunzmann) [reply hanging_threads.py:141]: You can create an issue: refactor monitor function and an issue to write tests.

Comment (artemrizhov) [reply hanging_threads.py:141]: Created #40 and #41.

**Pull request merged**